lib/charms/opensearch/v0/opensearch_config.py

```diff
@@ -191,6 +191,42 @@ def cleanup_bootstrap_conf(self):
         """Remove some conf entries when the cluster is bootstrapped."""
         self._opensearch.config.delete(self.CONFIG_YML, "cluster.initial_cluster_manager_nodes")
 
+    def get_plugin(self, plugin_config: Dict[str, str]) -> Dict[str, Any]:
+        """Gets a list of configuration from opensearch.yml."""
+        result = {}
+        loaded_configs = self.load_node()
+        for key in plugin_config.keys():
+            result[key] = loaded_configs.get(key, None)
+        return result
+
+    def add_plugin(self, plugin_config: Dict[str, str]) -> bool:
+        """Adds plugin configuration into opensearch.yml.
+
+        Returns True if all the configurations have been set correctly.
+        """
+        if not plugin_config:
+            return False
+        return all(
+            [
+                self._opensearch.config.put(self.CONFIG_YML, key, val).get(key, None) == val
```

Comment:
This expression looks quite complex and hard to understand tbh. 
Maybe it is better to split this into more steps for the sake of readability?